ui/src/actions/login.js

```diff
@@ -59,90 +83,205 @@ export function sendSelectProposal(number, navigate) {
   };
 }
 
-export function startSession(userInControl) {
-  return (dispatch) => {
-    dispatch(getInitialState(userInControl));
-    dispatch(setLoading(false));
+export function getLoginInfo() {
+  return async (dispatch) => {
+    const loginInfo = await fetchLoginInfo();
+    dispatch(setLoginInfo(loginInfo));
   };
 }
 
-export function getLoginInfo() {
-  return (dispatch) =>
-    fetchLoginInfo().then(
-      (loginInfo) => {
-        dispatch(setLoginInfo(loginInfo));
-        return loginInfo;
-      },
-      () => {
-        dispatch(showErrorPanel(true));
-        dispatch(setLoading(false));
-      },
-    );
-}
+export function logIn(proposal, password) {
+  return async (dispatch) => {
+    const res = await sendLogIn(proposal, password);
 
-export function logIn(proposal, password, navigate) {
-  // eslint-disable-next-line sonarjs/cognitive-complexity
-  return (dispatch) => {
-    const previousUser = localStorage.getItem('currentUser');
-    if (serverIO.hwrSocket !== null && serverIO.hwrSocket.connected) {
-      console.log(serverIO.hwrSocket.connected); // eslint-disable-line no-console
-    } else {
-      serverIO.connect();
+    if (res.msg !== '') {
```

Comment:
Ah, maybe that is the reason why all the success messages were removed on the back end side.

I am not convinced it is a great choice, but I can understand...

Empty message or not, the back end tests should test the message, because as we see here it is obviously part of the API.

Reply:
Ha, should have scrolled down further... Yeah, so the idea will be to return the login info object directly in a future PR.

Reply:
Ah, I see. Then @marcus-oscarsson I recommend to adjust the Python tests to assert that we do have an empty string where we expect one (as mentioned in earlier comments).